src/networking.c

```diff
@@ -1232,14 +1236,49 @@ int clientHasPendingReplies(client *c) {
 
 /* Return true if client connected from loopback interface */
 int islocalClient(client *c) {
+    struct ifaddrs *ifa, *addrs;
+    struct sockaddr_in *s4;
+    struct sockaddr_in6 *s6;
+    void *in_addr;
+    char loip[NET_IP_STR_LEN + 1];
+
     /* unix-socket */
     if (c->flags & CLIENT_UNIX_SOCKET) return 1;
 
     /* tcp */
     char cip[NET_IP_STR_LEN+1] = { 0 };
     connAddrPeerName(c->conn, cip, sizeof(cip)-1, NULL);
 
-    return !strcmp(cip,"127.0.0.1") || !strcmp(cip,"::1");
+    if (getifaddrs(&addrs) != 0) return 0;
+    for (ifa = addrs; ifa != NULL; ifa = ifa->ifa_next) {
+        if (!ifa->ifa_addr ||
+            !(ifa->ifa_flags & IFF_UP) ||
```

Comment:
Nit
```suggestion
            !((ifa->ifa_flags & MASK_LOOPBACK) == MASK_LOOPBACK)
```
where
```C
#define MASK_LOOPBACK (IFF_UP|IFF_LOOPBACK|IFF_RUNNING)
```

Reply:
@foudfou Thanks for your suggestion. Added the constant as `NETWORK_LOOPBACK_MASK`.